### DISCUSSION TITLE
Bug: defaultSelected doesn't  update if the value changes after the first render

### DISCUSSION THREAD

**Discussion by markcnunes** [gpbl/react-day-picker#1747] [Support] [answered]
### Bug description

defaultSelected from useInput doesn't update if the value changes after the first render.
This is an issue if the data is being fetched because it starts as undefined and finishes as a date but useInput doesn't update the defaultSelected.

### To reproduce

https://codesandbox.io/s/error-useinput-from-react-day-picker-hlspx8

#### Steps

1. Initiate useInput with defaultSelected as undefined 
2. Change defaultSelected to a Date 

### Expected behavior

It would make sense if useInput allowed defaultSelected to be modified.

### Additional context

The input field is undefined if the value changes after the first render
<img width="326" alt="image" src="https://user-images.githubusercontent.com/18724600/225362730-abafc194-8963-4c84-a693-c038164937d8.png">

It works fine when the date is provided in the first render
<img width="282" alt="image" src="https://user-images.githubusercontent.com/18724600/225363071-b7e5b220-43b7-4f4a-968d-795b0641fb5c.png">

I am currently having to overwrite the value for the input tag to be able to allow the input to display the right value after the first render.
```tsx
  <Input
          {...inputProps}
          value={value ? new Date(value) : undefined}
```

**Comment by san70sh** [comment]
I am facing the same issue and would love to see a resolution.

**Comment by gpbl** [comment, marked as answer]
@markcnunes please use `setSelected` returned by the hook to set the selected day:

https://codesandbox.io/s/wp2emv

**Reply by markcnunes** [reply]
Hi @gpbl , Thanks for the answer. It makes sense to use `setSelected` since `defaultSelected` shouldn't change. Having that in mind I just decided to delay rendering the component if I know that the initial value is coming from an API call e.g. `{!apiCall.isInitialLoading && <DatePicker value={apiCall.data.date} /> )}`
